test/controllers/samples_controller_test.rb

```diff
@@ -349,4 +349,118 @@ class SamplesControllerTest < ActionDispatch::IntegrationTest
       }
     end
   end
+
+  test 'report_info should return cached copy on second request' do
+    post user_session_path, params: @user_params
+
+    url = report_info_url
+    get url
+
+    assert_response :success
+
+    first_runtime = @response.headers["X-Runtime"]
+    cache_header = @response.headers["X-IDseq-Cache"]
+    assert_equal "missed", cache_header
+
+    get url
+    assert_response :success
+
+    second_runtime = @response.headers["X-Runtime"]
+    cache_header = @response.headers["X-IDseq-Cache"]
+    assert_equal "requested", cache_header
+
+    # Second request should always be faster than the first.
+    # Typically it will be 10x faster.
+    assert first_runtime.to_f > second_runtime.to_f
+  end
+
+  test 'report_info should override background when background is not viewable' do
+    post user_session_path, params: @user_params
+    # ids beyond any conceivable range
+    url = report_info_url(background_id: rand(10**10..11**10))
+    get url
+
+    assert_response :success
+  end
+
+  test 'report_info should return last-modified' do
+    post user_session_path, params: @user_params
+
+    report_ts = Time.now.utc
+    url = report_info_url(report_ts: report_ts.to_i)
+    get url
+    last_modified = Time.httpdate(@response.headers["Last-Modified"])
+    # Last-Modified of test data will be creation time, so we just match to the day
```

Comment:
This assertion is unstable if the test is running at midnight, that might give us false negatives.

Consider stubbing Time.now using ActiveSupport [TimeHelper](https://api.rubyonrails.org/classes/ActiveSupport/Testing/TimeHelpers.html#method-i-freeze_time):

```ruby
test 'report_info should return last-modified' do
   freeze_time do
      report_ts = Time.now.utc
      
      post user_session_path, params: @user_params

      url = report_info_url(report_ts: report_ts.to_i)
      get url
      last_modified = Time.httpdate(@response.headers["Last-Modified"])

      # now you can assert the entire date
   end

```

Reply:
Nice, I didn't know about `freeze_time` 😄